### PR TITLE
added tutorial, themed docs, added README.rst for PyPi

### PR DESCRIPTION
**Pull request by ryochiji**
* Use a nicer theme for Sphinx docs
 * Add codelab/tutorial 
 * Minor changes to `make docs` so customized .rst files aren't blown away
 * Added PyPi-friendly README.rst (they don't render markdown correctly)

image

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -37,12 +37,15 @@ pypi_upload: install
 	$(call header,"Uploading new version to PyPi")
 	@. $(INSTALL_DIR)/bin/activate; python setup.py sdist upload -r pypi
 
-docs: install
-	$(call header,"Building docs")
+clean_docs: install
```
Comment (aleontiev) [inline Makefile]: Running this would nuke the tutorial, right? When would we want to do this?

**Comment by aleontiev** [comment]
:+1: looks great! One question about the purpose of `make clean_docs`

File: docs/_sources/tutorial.txt [hunk 2]
```diff
@@ -0,0 +1,458 @@
...
+
+Before starting the tutorial, you’ll need to set up Dynamic REST in your dev environment. Refer to the README for details, but it should look something like::
+
+    $ git clone git@github.com:AltSchool/dynamic-rest.git
+    $ cd dynamic_rest
+    $ CMD=migrate make run
+    $ CMD=shell make run
+    >>> from tests.setup import create_fixture
+    >>> create_fixture()
+    $ make server
```
Comment (aleontiev) [inline docs/_sources/tutorial.txt]: We can probably add a make wrapper to do all of this, similar to what is currently happening in the Procfile: `manage.py migrate --settings=tests.settings && manage.py runserver 0.0.0.0:$PORT --settings=tests.settings`

File: docs/_sources/tutorial.txt [hunk 3]
```diff
@@ -0,0 +1,458 @@
...
+
+            return super(EventViewSet, self).list(request, *args, **kwargs)
+
+Checkout the default results:
+http://localhost:9002/events/
+
+
+Queryset override
+-----------------
+By default, DREST/DRF will query the model declared in the viewset’s serializer, which is to say, all objects in that model are in-scope and query-able. If you want to change that, you can override the ``get_queryset()`` method in your viewset. One real-life example is in the ALO Competencies API, where we switch the default queryset to an MPTT queryset constrained to the descendants of a root node.
```
Comment (aleontiev) [inline docs/_sources/tutorial.txt]: The ALO Competencies API.. what's that :)

File: docs/_sources/tutorial.txt [hunk 4]
```diff
@@ -0,0 +1,458 @@
...
+
+    class BlogPostSerializer(DynamicModelSerializer):
+        # ...
+
+        author = DynamicRelationField(UserSerializer, embed=True)
+
+
+Default Filtering of Related Objects
+------------------------------------
+DynamicRelationFields can have a default queryset/filter. While clients can apply filtering on related objects (and viewsets can do the same through query injection), sometimes a default filter needs to be applied in all cases, and you don’t want to leave it to the client to know that. An example might be a classroom.students relation, where students should always be filtered by enrollment status, and expecting a client who does include[]=teacher.classroom.students to also apply filter{teacher.classroom|student.enrolled}=1 is burdensome and error-prone.
```
Comment (aleontiev) [inline docs/_sources/tutorial.txt]: Maybe use more generic resource types here (events, locations, etc)

File: Makefile (hunk 1 above)
Comment (ryochiji) [inline Makefile]: I'm ok with removing it. I guess if we ever needed a complete reset we can just run the commands manually.

File: docs/_sources/tutorial.txt (hunk 3 above)
Comment (ryochiji) [inline docs/_sources/tutorial.txt]: Good catch

File: Makefile (hunk 1 above)
Comment (aleontiev) [inline Makefile]: Yeah.. the nice thing about this is that as we change our docstrings, running `make clean_docs` would update the documentation. 

In a future change, we can probably figure out a strategy by which we temporarily move custom documentation out of the `docs` folder, then remove the docs folder, re-generate the documentation, and move the custom documentation back in to `docs`.

**Comment by ryochiji** [comment]
@aleontiev - updated

**Comment by aleontiev** [comment]
:+1:

**Pull request merged**